src/modules/Api.ts

```diff
@@ -183,6 +189,107 @@ export class Api {
     });
   }
 
+  public async canonicalOrder({
+    order: {
+      side,
+      market,
+      amount,
+      limitPrice,
+      isDecreaseOnly = false,
+      signedTriggerPrice = '0',
+      makerAccountNumber = '0',
+      makerAccountOwner,
+      expiration  = new BigNumber(FOUR_WEEKS_IN_SECONDS),
+      limitFee,
+    },
+    triggerPrice = '0',
+    fillOrKill,
+    postOnly,
+    marginDeposit,
+    payoutAmount,
+    clientId,
+    setExpirationOnFill,
+    cancelId,
+    cancelAmountOnRevert,
+  }: {
+    order: {
+      side: ApiSide,
+      market: ApiMarketName,
+      amount: Integer | string,
+      limitPrice: string,
+      isDecreaseOnly: boolean,
+      signedTriggerPrice: string,
+      makerAccountNumber: Integer | string,
+      makerAccountOwner: address,
+      expiration: Integer | string,
+      limitFee?: string,
+    },
+    triggerPrice: string,
+    fillOrKill?: boolean,
+    postOnly?: boolean,
+    marginDeposit?: Integer | string,
+    payoutAmount?: Integer | string,
+    clientId?: string,
+    setExpirationOnFill?: boolean,
+    cancelId?: string,
+    cancelAmountOnRevert?: boolean,
+  }): Promise<{ order: ApiOrder }> {
+    const realExpiration: BigNumber = getRealExpiration(expiration);
+
+    const markets: string[] = market.split('-');
+    const baseMarket: BigNumber = new BigNumber(markets[0]);
```

Comment:
markets[0] is a string. You will have to convert to BigNumber via the `MarketId` object in types.ts